### ISSUE TITLE
Plugin does not ignore commented-out `expect` statements or test exclusions with `.only`

### ISSUE THREAD

**Issue by huy-nguyen** [mattphillips/babel-jest-assertions#1]
**Bug**

- `babel-jest-assertions` version: 0.0.1
- `node` version: 8.3.0
- `npm` (or `yarn`) version: 5.3.0

Relevant code or config

```javascript
test('dummy', () => {
  // expect(1).toBe(1);
})
```

What you did: Run `jest` on the above test file.

What happened (please provide anything you think will help):`jest-babel-assertions` incorrectly inserts `expect.asertions` for the commented out `expect`, causing the test to fail:

```
expect.hasAssertions()

    Expected at least one assertion to be called but received none.
```
Another file that causes the same problem:
```javascript
test('does not run', () => {
  expect(1).toBe(1);
})
test.only('runs', () => {
  expect(1).toBe(1)
})
```
In this case the error is on the `does not run` test.

Reproduction repository (if possible):

**Comment by mattphillips** [comment]
Hey @huy-nguyen thanks for raising this issue and nice catch!

I'm on holiday at the moment and don't have a computer with me 😢 so I won't be able to fix this right now.

How would you feel about sending a PR with the fix for the commented out expects? 

The basic logic of this plugin is that it turns the test into a string and matches on all `expect(` patterns found to determine the count. What we need to change is the code string being used to do the match on. 

On line 42 in src/index.js. You would need to put a normalised version of `code` variable by removing all comments in the code string. Once normalised we would use the new variable to determine the count, hasAssertions and expect.assertions already exists.

```js
const normalisedCode = code.replace(/\/\/(.*)/g, '').replace(/\/\*([\s\S]*)\*\//g, '');
```

If you could add test cases for this with the following formats that would be great.

```js
test('returns 2 when given 1 and 1', () => {
  /*
  expect(1).toBe(1);
  */
  const a = 1; /* expect(a).toBe(1) */
  const b = 1; // expect(b).toBe(1)
  expect(add(a, b)).toBe(2);
  // expect (add(1, 2)).toBe(3);
})

```

If you don't have time to send a PR no worries dude just let me know and I'll try and find someone who can help in my absence 😄

**Comment by mattphillips** [comment]
As for the second issue with `.only` tests, this appears to be an issue in jest [see](https://github.com/facebook/jest/issues/3595). Until this is fixed the hasAssertions will break in this plugin, perhaps we should remove it until then?

**Comment by hiddentao** [comment]
PR in #2

**Comment by mattphillips** [comment]
Thanks @huy-nguyen for the issue and @hiddentao for the PR. Now got to figure out how to release without a computer 😂

Once I've published the fix I'll let you know.

**Comment by mattphillips** [comment]
Re-opening for the issue with focussed tests and Jests hasAssertions

**Comment by mattphillips** [comment]
@huy-nguyen Jest hasAssertions has been fixed in the above PR, just waiting on a release of Jest.

**Comment by mattphillips** [comment]
I've just published v0.0.2 with the fix for commented out expects and the issue in jest should be released within a week 🎉
